File: frontend/src/services/inventoryService.ts
import axiosInstance from '../api/axiosInstance';
import type { Inventory } from '../types/inventory';

const getInventories = () => axiosInstance.get<Inventory[]>('/inventories').then(res => res.data);
const getInventory = (id: string) => axiosInstance.get<Inventory>(`/inventories/${id}`).then(res => res.data);
const adjustInventory = (inventoryId: string, quantity: number, remark?: string) =>
  axiosInstance.post<unknown>(`/inventories/adjust?inventoryId=${inventoryId}&quantity=${quantity}${remark ? `&remark=${encodeURIComponent(remark)}` : ''}`);
const getTransactionsByBranch = (branchId: string) =>
  axiosInstance.get(`/inventories/${branchId}/transactions`).then(res => res.data);

export default { getInventories, getInventory, adjustInventory, getTransactionsByBranch };

File: backend/src/main/java/com/pos/service/InventoryService.java
package com.pos.service;

import com.pos.entity.Inventory;
import com.pos.entity.InventoryTransaction;

import java.util.List;
import java.util.UUID;

public interface InventoryService {
    List<Inventory> getAllInventories();
    Inventory getInventoryById(UUID id);
    InventoryTransaction adjustInventory(UUID inventoryId, int quantity, String remark);
    List<InventoryTransaction> getTransactionsByBranch(UUID branchId);
}
